Gems/ROS2/Code/Include/ROS2/Utilities/ROS2Conversions.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Math/Vector3.h>
+#include <geometry_msgs/msg/point.hpp>
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+
+namespace ROS2
+{
+    //! Utility class for conversions between ROS2 types and O3DE (AZ::) types.
+    class ROS2Conversions
+    {
+    public:
+        static AZ::Vector3 FromROS2Vector3(const geometry_msgs::msg::Vector3& ros2vector);
+        static geometry_msgs::msg::Vector3 ToROS2Vector3(const AZ::Vector3& azvector);
+        static geometry_msgs::msg::Point ToROS2Point(const AZ::Vector3& azvector);
+        static geometry_msgs::msg::Quaternion ToROS2Quaternion(const AZ::Quaternion& azquaternion);
+        static geometry_msgs::msg::Pose ToROS2Pose(const AZ::Transform& aztransform);
+        static AZ::Transform FromROS2Pose(const geometry_msgs::msg::Pose& ros2pose);
+        static AZ::Vector3 FromROS2Point(const geometry_msgs::msg::Point& ros2point);
+        static AZ::Quaternion FromROS2Quaternion(const geometry_msgs::msg::Quaternion& ros2quaternion);
+    };
```

Comment:
I recommend using a namespace over a class with static functions.

The reason why is that with a namespace the functions can be forwarded declared without the need to include this header, while a class with static functions can't be as the class definition is needed for the functions.

i.e The following works when using a namespace

#### SomeOtherSourceFileThatDoesNotIncludeRos2Conversions.cpp
---
```c++
namespace ROS2::ROS2Conversions
{
    AZ::Vector3 FromROS2Vector3(const geometry_msgs::msg::Vector3& ros2vector);
}
```